app/main.py

```diff
@@ -21,38 +21,24 @@ def __iadd__(self, other):
         return self
 
     def __mul__(self, other):
-        if isinstance(other, Distance):
-            return Distance(self.km * other.km)
         return Distance(self.km * other)
 
     def __truediv__(self, other):
-        if isinstance(other, Distance):
-            return Distance(round((self.km / other.km), 2))
         return Distance(round((self.km / other), 2))
 
     def __lt__(self, other):
-        if isinstance(other, Distance):
-            return self.km < other.km
         return self.km < other
 
     def __gt__(self, other):
-        if isinstance(other, Distance):
-            return self.km > other.km
         return self.km > other
 
     def __eq__(self, other):
-        if isinstance(other, Distance):
-            return self.km == other.km
         return self.km == other
```

Comment:
you spoiled the correct implementation :( You still need to do a type check here